packages/account/src/test-utils/setup-test-provider-and-wallets.test.ts

```diff
@@ -0,0 +1,263 @@
+import { randomBytes } from '@fuel-ts/crypto';
+import { ErrorCode } from '@fuel-ts/errors';
+import { expectToThrowFuelError, safeExec } from '@fuel-ts/errors/test-utils';
+import type { AbstractAddress } from '@fuel-ts/interfaces';
+import { bn, toNumber } from '@fuel-ts/math';
+import { defaultSnapshotConfigs, hexlify } from '@fuel-ts/utils';
+import { waitUntilUnreachable } from '@fuel-ts/utils/test-utils';
+
+import { Provider } from '../providers';
+import { Signer } from '../signer';
+import { WalletUnlocked } from '../wallet';
+
+import { AssetId } from './asset-id';
+import * as launchNodeMod from './launchNode';
+import { setupTestProviderAndWallets } from './setup-test-provider-and-wallets';
+import { TestMessage } from './test-message';
+
+const BaseAssetId = `0x${defaultSnapshotConfigs.chainConfig.consensus_parameters.V1.base_asset_id}`;
+/**
+ * @group node
+ */
+describe('setupTestProviderAndWallets', () => {
+  it('kills the node after going out of scope', async () => {
+    let url = '';
+    // eslint-disable-next-line no-lone-blocks
+    {
+      using result = await setupTestProviderAndWallets();
+      url = result.provider.url;
+      await result.provider.getBlockNumber();
+    }
+
+    await waitUntilUnreachable(url);
+
+    const { error } = await safeExec(async () => {
+      const p = await Provider.create(url);
+      return p.getBlockNumber();
+    });
+
+    expect(error).toMatchObject({
+      message: 'fetch failed',
+    });
+  });
+
+  test('kills the node if provider cant connect post-launch', async () => {
```

Comment:
Why the variation between using `test` and `it` here?

Reply:
There's no real reason, I'd just write what comes to mind.